src/test/java/net/snowflake/client/jdbc/SnowflakeTypeTest.java

```diff
@@ -0,0 +1,118 @@
+    assertEquals(convertStringToType("byteint"), Types.INTEGER);
+    assertEquals(convertStringToType("smallint"), Types.SMALLINT);
+    assertEquals(convertStringToType("bigint"), Types.BIGINT);
+    assertEquals(convertStringToType("double"), Types.DOUBLE);
+    assertEquals(convertStringToType("double precision"), Types.DOUBLE);
+    assertEquals(convertStringToType("real"), Types.REAL);
+    assertEquals(convertStringToType("char"), Types.CHAR);
+    assertEquals(convertStringToType("character"), Types.CHAR);
+    assertEquals(convertStringToType("varbinary"), Types.VARBINARY);
+    assertEquals(convertStringToType("boolean"), Types.BOOLEAN);
+    assertEquals(convertStringToType("date"), Types.DATE);
+    assertEquals(convertStringToType("time"), Types.TIME);
+    assertEquals(convertStringToType("timestamp"), Types.TIMESTAMP);
+    assertEquals(convertStringToType("datetime"), Types.TIMESTAMP);
+    assertEquals(convertStringToType("timestamp_ntz"), Types.TIMESTAMP);
+    assertEquals(convertStringToType("timestamp_ltz"), Types.TIMESTAMP_WITH_TIMEZONE);
+    assertEquals(convertStringToType("timestamp_tz"), Types.TIMESTAMP_WITH_TIMEZONE);
+    assertEquals(convertStringToType("variant"), Types.OTHER);
+    assertEquals(convertStringToType("object"), Types.JAVA_OBJECT);
+    assertEquals(convertStringToType("vector"), SnowflakeUtil.EXTRA_TYPES_VECTOR);
+    assertEquals(convertStringToType("array"), Types.ARRAY);
+    assertEquals(convertStringToType("default"), Types.OTHER);
+  }
+
+  @Test
+  public void testJavaSQLTypeFind() {
+    assertNull(SnowflakeType.JavaSQLType.find(200000));
+  }
+
+  @Test
+  public void testJavaSQLTypeLexicalValue() {
+    assertEquals(SnowflakeType.lexicalValue(1.0f, null, null, null, null), "0x1.0p0");
+    assertEquals(SnowflakeType.lexicalValue(new BigDecimal(100.0), null, null, null, null), "100");
+    assertEquals(
+        SnowflakeType.lexicalValue("random".getBytes(), null, null, null, null), "72616E646F6D");
+  }
+
+  @Test
+  public void testJavaTypeToSFType() {
+    try {
+      assertEquals(SnowflakeType.javaTypeToSFType(0, null), SnowflakeType.ANY);
+      assertThrows(
+          SnowflakeSQLLoggedException.class,
+          () -> {
+            SnowflakeType.javaTypeToSFType(2000000, null);
+          });
+      ;
+    } catch (SnowflakeSQLException ex) {
+      fail("Failed the test");
+    }
+  }
+
+  @Test
+  public void testJavaTypeToClassName() {
+    try {
+      assertEquals(SnowflakeType.javaTypeToClassName(Types.DECIMAL), BigDecimal.class.getName());
+      assertEquals(SnowflakeType.javaTypeToClassName(Types.TIME), java.sql.Time.class.getName());
+      assertEquals(SnowflakeType.javaTypeToClassName(Types.BOOLEAN), Boolean.class.getName());
+      assertThrows(
+          SQLFeatureNotSupportedException.class,
+          () -> {
+            SnowflakeType.javaTypeToClassName(-2000000);
+          });
+    } catch (SQLException ex) {
```

Comment:
let's not catch the exception just to fail the test - it will be done by Junit (also in other places)

Reply:
I Removed the catch blocked.